experimental/dds/tree2/src/feature-libraries/modular-schema/fieldChangeHandler.ts

```diff
@@ -183,6 +183,10 @@ export type NodeChangeRebaser = (
 	 * True when the baseChange deletes the node. Defaults to false if not specified.
 	 */
 	deleted?: boolean,
+	/**
+	 * True when the baseChange revives the node. Defaults to false if not specified.
+	 */
+	revived?: boolean,
```

Comment:
It seems a bit weird that this is a separate parameter from `deleted` since they're not independent. Could we instead have a single enum which can represent deleted, revived, and unchanged?

Reply:
Created `NodeExistenceStateChange` for this now. I also pulled some of the conditionals in `rebaseNodeChange` out to `existenceConstraintDelta` as I thought it was getting difficult to read. If we don't like this I can move it back in/do it differently.